prov/verbs/src/verbs_domain.c

```diff
@@ -351,11 +351,11 @@ fi_ibv_domain(struct fid_fabric *fabric, struct fi_info *info,
 
 	if (fi->nic && fi->nic->device_attr &&
 	    !strncmp(fi->nic->device_attr->vendor_id, "0x02c9", 6)) {
-		_domain->post_send = ibv_post_send;
-		_domain->poll_cq = ibv_poll_cq;
-	} else {
 		_domain->post_send = fi_ibv_post_send_track_credits;
 		_domain->poll_cq = fi_ibv_poll_cq_track_credits;
+	} else {
+		_domain->post_send = ibv_post_send;
+		_domain->poll_cq = ibv_poll_cq;
```

Comment:
The intent was to have Mellanox NICs take the optimized path. Other NICs would use the path that adds the credit tracking.